File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution(object):
    def longestSubarray(self, nums):
        """
        must delete one item 
        
        [0,1,1,1,0,1,1,0,1]
        
        
        """
        result = 0
        left = 0
        ziro_counter = 0
        
        for right in range(len(nums)):
            if nums[right] == 0:
                ziro_counter += 1
            while ziro_counter > 1:
                if nums[left] == 0:
                    ziro_counter -= 1
                left += 1
            result = max(result, right - left)
            
        return result

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.js
var longestSubarray = function(nums) {
   let result = 0
    let left = 0
    let ziro_counter = 0
        
    for(let right=0; right<nums.length; right++){
        if (nums[right] == 0){
            ziro_counter += 1
        }    
            while( ziro_counter > 1){
                if (nums[left] == 0){
                    ziro_counter -= 1
                }   
                left += 1
            }
            result = Math.max(result, right - left)
    }
                  
        return result 
};